src/Indice.Features.Cases.AspNetCore/Services/BaseCaseService.cs

```diff
@@ -159,6 +162,11 @@ internal abstract class BaseCaseService
             AssignedTo = assignee == null ? null : AuditMeta.Create(assignee)
         };
 
+        // If enabled get a new reference number from the sequence.
+        if (_options.EnableReferenceNumber) {
+            entity.ReferenceNumber = await _dbContext.NextReferenceNumber();
```

Comment:
βαλε εδω την λογικη του next value for, και δοκιμασε να μην βαλεις όλα τα namespaces αλλά απ΄ ευθειας `select next value for ReferenceNumberSequence`. Επειδη ήδη θα έχει σηκωσει το dbContext πιθανοτατα να μην χρειαζοντα τα πάντα (ωστε να θελεις και λιγοτερα dependencies)